### PR TITLE
MPDX-8383 - Complete Task Modal - preselect result

### PR DESCRIPTION
**Pull request by wjames111**
## Description

- Jira Ticket [8383](https://jira.cru.org/browse/MPDX-8383)
- Description: In the complete task modal, if there is only one task result option, we can preselect it to speed up completing tasks for users.

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [x] I have requested a review from another person on the project

### CODE REVIEW

File: src/components/Task/Modal/Form/Inputs/ResultAutocomplete/ResultAutocomplete.test.tsx [hunk 1]
```diff
@@ -84,4 +84,11 @@ describe('ResultSelect', () => {
       phaseData,
     });
   });
+  it('Preselects result when only one result available', () => {
```
Comment (dr-bizz) [inline src/components/Task/Modal/Form/Inputs/ResultAutocomplete/ResultAutocomplete.test.tsx:87]:
```suggestion
  it('preselects a result when only one result is available', () => {
```

**Review by dr-bizz — APPROVED** [review]
This is great! Thank you for fixing it so fast!

**Review by canac — CHANGES_REQUESTED** [review]
I don't think this is quite right. It shows the correct value in the UI but doesn't seem to actually select it because `displayResult` is being sent as `null`.

**Comment by canac** [comment]
@wjames111 Yeah, I think that's exactly what you'll want to do.

File: src/components/Task/Modal/Form/Inputs/ResultAutocomplete/ResultAutocomplete.test.tsx [hunk 2]
```diff
@@ -84,4 +84,11 @@ describe('ResultSelect', () => {
       phaseData,
     });
   });
+  it('preselects a result when only one result is available', () => {
+    const { getByRole } = render(
+      <Components availableResults={[ResultEnum.Attempted]} />,
+    );
+    expect(getByRole('combobox', { name: 'Result' })).toBeInTheDocument();
+    expect(getByRole('combobox', { name: 'Result' })).toHaveValue('Attempted');
```
Comment (canac) [inline src/components/Task/Modal/Form/Inputs/ResultAutocomplete/ResultAutocomplete.test.tsx:92]: The `getByRole` for the value check is enough to also assert that it is in the document.

```suggestion
    expect(getByRole('combobox', { name: 'Result' })).toHaveValue('Attempted');
```

File: src/components/Task/Modal/Form/TaskModalHelper.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ export const handleTaskPhaseChange = ({
   setFieldValue('taskPhase', phase);
   setFieldValue('activityType', '');
   setFieldValue('subject', '');
-  setFieldValue('displayResult', undefined);
+  setFieldValue('displayResult', '');
```
Comment (canac) [inline src/components/Task/Modal/Form/TaskModalHelper.ts:66]: I think this change is causing an error when I try to add or edit a task:

```
Variable $attributes of type TaskUpdateInput! was provided invalid value for displayResult (Expected "" to be one of: INITIATION_RESULT_NO_RESPONSE, INITIATION_RESULT_CIRCLE_BACK, INITIATION_RESULT_APPOINTMENT_SCHEDULED, INITIATION_RESULT_NOT_INTERESTED, FOLLOW_UP_RESULT_NO_RESPONSE, FOLLOW_UP_RESULT_PARTNER_FINANCIAL, FOLLOW_UP_RESULT_PARTNER_SPECIAL, FOLLOW_UP_RESULT_PARTNER_PRAY, FOLLOW_UP_RESULT_NOT_INTERESTED, APPOINTMENT_RESULT_CANCELLED, APPOINTMENT_RESULT_FOLLOW_UP, APPOINTMENT_RESULT_PARTNER_FINANCIAL, APPOINTMENT_RESULT_PARTNER_SPECIAL, APPOINTMENT_RESULT_PARTNER_PRAY, APPOINTMENT_RESULT_NOT_INTERESTED, PARTNER_CARE_COMPLETED)
```

**Review by canac — CHANGES_REQUESTED** [review]
Almost there!

**Comment by canac** [comment]
@wjames111 I think we can fix the bug where the result is populated before the user picks an action by passing an empty array to `availableResults` when the `activityType` is missing.

Sorry, I thought you had requested another review. I'm sure you were going to get to that bug, but I'm just mentioning what I found since I already spent the time looking at the code.

The rest looks great once you get all the tests passing again!

```patch
diff --git a/src/components/Task/Modal/Form/Complete/TaskModalCompleteForm.tsx b/src/components/Task/Modal/Form/Complete/TaskModalCompleteForm.tsx
index 7b38566fc..d572c847a 100644
--- a/src/components/Task/Modal/Form/Complete/TaskModalCompleteForm.tsx
+++ b/src/components/Task/Modal/Form/Complete/TaskModalCompleteForm.tsx
@@ -350,7 +350,7 @@ const TaskModalCompleteForm = ({
               </Grid>
 
               <ResultAutocomplete
-                availableResults={availableResults}
+                availableResults={activityType ? availableResults : []}
                 result={displayResult}
                 setFieldValue={setFieldValue}
                 setResultSelected={setResultSelected}
diff --git a/src/components/Task/Modal/Form/LogForm/TaskModalLogForm.tsx b/src/components/Task/Modal/Form/LogForm/TaskModalLogForm.tsx
index 98babc1e2..fcf96c4e4 100644
--- a/src/components/Task/Modal/Form/LogForm/TaskModalLogForm.tsx
+++ b/src/components/Task/Modal/Form/LogForm/TaskModalLogForm.tsx
@@ -421,7 +421,7 @@ const TaskModalLogForm = ({
               )}
 
               <ResultAutocomplete
-                availableResults={availableResults}
+                availableResults={activityType ? availableResults : []}
                 result={displayResult}
                 setFieldValue={setFieldValue}
                 setResultSelected={setResultSelected}
diff --git a/src/components/Task/Modal/Form/TaskModalForm.tsx b/src/components/Task/Modal/Form/TaskModalForm.tsx
index fbf817d9b..aa4920862 100644
--- a/src/components/Task/Modal/Form/TaskModalForm.tsx
+++ b/src/components/Task/Modal/Form/TaskModalForm.tsx
@@ -583,7 +583,7 @@ const TaskModalForm = ({
               </Grid>
               {initialTask.completedAt && (
                 <ResultAutocomplete
-                  availableResults={availableResults}
+                  availableResults={activityType ? availableResults : []}
                   result={displayResult}
                   setFieldValue={setFieldValue}
                   setResultSelected={setResultSelected}
```

**Review by canac — APPROVED** [review]
This is great! As a user, I'm looking forward to saving some clicks when I complete tasks!

**Pull request merged**